components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/tokenprocessor/HashingPersistenceProcessor.java

```diff
@@ -104,6 +105,10 @@ public String getPreprocessedRefreshToken(String processedRefreshToken) throws I
      */
     private String hash(String plainText) throws IdentityOAuth2Exception {
 
+        if (StringUtils.isEmpty(plainText)) {
+            throw new IllegalArgumentException("Invalid plainText value to be hash.");
```

Comment:
@Biruntha shall we change this error message to inform that the `plainText` is empty or null other than just mentioning invalid?